core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -11590,6 +11590,122 @@ public void testAcquisitionNotThrottledIfHighDeliveryCountRecordNotAcquired() {
         assertArrayEquals(expectedAcquiredRecords.toArray(), acquiredRecordsList.toArray());
     }
 
+    @Test
+    public void testAcquisitionThrottlingWithOnGoingStateTransition() {
```

Comment:
nit: Ongoing is a word, so `testAcquisitionThrottlingWithOngoingStateTransition` is slightly better.